### PR TITLE
Fixed style and consistency

### PR DESCRIPTION
**Pull request by SyluxImmortal**
Added tslint and @types/screeps
Added tslint to build process
Removed old typings
Separated several interfaces into their own files

### CODE REVIEW

File: examples/bundles/coop.bundle.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+// tslint:disable-next-line:class-name
 class PosisTest_CooperativeSchedulingProcess implements IPosisProcess {
 
     public static ImageName = "PosisTest/CooperativeSchedulingProcess";
 
     protected coop: IPosisCooperativeScheduling;
 
     public constructor(context: IPosisProcessContext) {
-        this.coop = shimCooperativeScheduling(context.queryPosisInterface('coop'));
+        this.coop = shimCooperativeScheduling(context.queryPosisExtension('coop'));
```
Comment (AlinaNova21) [inline examples/bundles/coop.bundle.ts]: This shouldn't be changed in this PR ;)

**Comment by AlinaNova21** [comment]
LGTM

**Comment by SyluxImmortal** [comment]
So the bot just closes stuff after awhile?

**Comment by AlinaNova21** [comment]
Apparently after 7 daysof inactivity
https://github.com/screepers/POSIS/blob/master/.gitconsensus.yaml#L19-L20

**Comment by AlinaNova21** [comment]
Reopening and merging as this technically doesn't change anything other than a bit of cleanup and styling consistency.

**Pull request merged**